backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/CpuPinningHelper.java

```diff
@@ -100,26 +102,93 @@ private static List<Integer> createRange(int start, int end) {
     }
 
     /**
-     * Provides a CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
+     * Provides a dedicated CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
      *
      * @param vdsCpuUnits a list of VdsCpuUnit
      * @return a map of CPU pinning
      */
-    public static Map<Integer, Integer> createCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
+    public static Map<Integer, Integer> createDedicatedCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
         return IntStream.range(0, vdsCpuUnits.size())
                 .boxed()
                 .collect(Collectors.toMap(Function.identity(), i -> vdsCpuUnits.get(i).getCpu()));
     }
 
+    /**
+     * Provides a CPU pinning map based on the VdsCpuUnit list and the CPU pinning policy. The pinning will be based on the list order.
+     * @param vdsCpuUnits a list of VdsCpuUnit
+     * @param cpuPinningPolicy the CPU pinning policy
+     * @return a map of CPU pinning
+     */
+    public static Map<Integer, Integer> createExclusiveCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits, CpuPinningPolicy cpuPinningPolicy) {
+        switch(cpuPinningPolicy) {
+            case DEDICATED:
+                return createDedicatedCpuPinningMap(vdsCpuUnits);
+            case ISOLATE_THREADS:
+                return createIsolatedThreadsCpuPinningMap(vdsCpuUnits);
+        }
+        return null;
+    }
+
+    /**
+     * Provides an isolate threads CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
+     *
+     * @param vdsCpuUnits a list of VdsCpuUnit
+     * @return a map of CPU pinning
+     */
+    private static Map<Integer, Integer> createIsolatedThreadsCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
+        List<Integer> socketsUsed = vdsCpuUnits.stream().map(VdsCpuUnit::getSocket).distinct().collect(Collectors.toList());
+        Map<Integer, Integer> mapping = new HashMap<>();
+        int vcpu = 0;
+        for (int socket : socketsUsed) {
+            List<VdsCpuUnit> cpusInSocket = vdsCpuUnits.stream().filter(cpu -> cpu.getSocket() == socket).collect(Collectors.toList());
+            List<Integer> cores = cpusInSocket.stream().map(VdsCpuUnit::getCore).distinct().collect(Collectors.toList());
+            for (int core : cores) {
+                mapping.put(vcpu++, cpusInSocket
+                        .stream()
+                        .filter(cpu -> cpu.getCore() == core)
+                        .map(VdsCpuUnit::getCpu)
+                        .findFirst().orElse(null));
```

Comment:
I do not like putting the `null` into the map. I think it should actually never happen, in that case it might be better to throw an exception to make it explicit that it is an illegal state.